### PR TITLE
front: fix linter warnings on generated API client

### LINKED ISSUE #3355: make linter compliant openApi files generate automatically
When `src/common/api/osrdMiddlewareApi.ts` `src/common/api/osrdEditoastApi.ts` and are generated, they are not properly styled.

These files should comply with our linter rules.

### PR DESCRIPTION
**Pull request by anisometropie**
fixes #3355

### CODE REVIEW

File: front/package.json [hunk 1]
```diff
@@ -165,8 +165,11 @@
     "start": "import-meta-env-prepare -x .env.example && vite",
     "build": "vite build",
     "test": "import-meta-env-prepare -x .env.example && vitest",
+    "lint": "eslint --ext .ts,.tsx,.js,.jsx src",
     "test-coverage": "import-meta-env-prepare -x .env.example && vitest run --coverage",
-    "generate-types": "npx @rtk-query/codegen-openapi src/config/openapi-middleware-config.ts && npx @rtk-query/codegen-openapi src/config/openapi-editoast-config.ts",
+    "generate-types": "yarn generate-types-main && yarn generate-types-eslint",
+    "generate-types-main": "npx @rtk-query/codegen-openapi src/config/openapi-middleware-config.ts && npx @rtk-query/codegen-openapi src/config/openapi-editoast-config.ts",
+    "generate-types-eslint": "eslint --fix src/common/api/osrdMiddlewareApi.ts src/common/api/osrdMiddlewareApi.ts",
```
Comment (multun) [inline front/package.json]: can you merge these 3 tasks into one?

**Review by multun — CHANGES_REQUESTED** [review]

File: front/package.json (hunk 1 above)
Comment (anisometropie) [reply front/package.json]: I wanted to avoid having so much shit crammed in one single line

File: front/package.json [hunk 2]
```diff
@@ -165,8 +165,11 @@
     "start": "import-meta-env-prepare -x .env.example && vite",
     "build": "vite build",
     "test": "import-meta-env-prepare -x .env.example && vitest",
+    "lint": "eslint --ext .ts,.tsx,.js,.jsx src",
```
Comment (multun) [inline front/package.json]: another PR with a lint task was merged, please remove this one

File: front/package.json (hunk 1 above)
Comment (multun) [reply front/package.json]: please use a post script then: https://docs.npmjs.com/cli/v6/using-npm/scripts#pre--post-scripts
you could also move it all into a shell script, and replace it all with `exec generate-types.sh`

**Review by multun — CHANGES_REQUESTED** [review]

File: front/package.json (hunk 1 above)
Comment (anisometropie) [reply front/package.json]: done ! I like the idea of using of script, much cleaner.

File: front/package.json (hunk 2 above)
Comment (anisometropie) [reply front/package.json]: Done, I just moved it to another line.

File: front/package.json [hunk 3]
```diff
@@ -166,11 +166,11 @@
     "start": "import-meta-env-prepare -x .env.example && vite",
     "build": "vite build",
     "test": "import-meta-env-prepare -x .env.example && vitest",
+    "lint": "eslint --ext .ts,.tsx,.js,.jsx src --max-warnings 345",
     "test-coverage": "import-meta-env-prepare -x .env.example && vitest run --coverage",
-    "generate-types": "npx @rtk-query/codegen-openapi src/config/openapi-middleware-config.ts && npx @rtk-query/codegen-openapi src/config/openapi-editoast-config.ts",
+    "generate-types": "yarn exec scripts/generate-types.sh",
```
Comment (multun) [inline front/package.json]:
```suggestion
    "generate-types": "exec scripts/generate-types.sh",
```
exec is a shell builtin, no need to use yarn for that

File: front/scripts/generate-types.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+npx @rtk-query/codegen-openapi src/config/openapi-middleware-config.ts
```
Comment (multun) [inline front/scripts/generate-types.sh:2]:
```suggestion
#!/bin/sh
npx @rtk-query/codegen-openapi src/config/openapi-middleware-config.ts
```

**Review by multun — CHANGES_REQUESTED** [review]

**Review by nicolaswurtz — APPROVED** [review]

**Review by multun — APPROVED** [review]

**Pull request merged**